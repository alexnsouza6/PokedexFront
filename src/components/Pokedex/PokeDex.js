import React, { Component } from "react";
import { Card, Image, Grid, List } from "semantic-ui-react";
import axios from "axios";
import "./style.css";

class PokeDex extends Component {
  // TO DO: put some margin: 0 auto at Image

  state = { pokemons: [] };

  async componentDidMount() {
    const response = await axios.get(
      `${process.env.REACT_APP_POKEDEX_API_URL}/pokemons`
    );
    this.setState({ pokemons: response.data });
  }

  handlePokemonSelect = pokemon => {
    this.props.history.push(`/pokemon/${pokemon.id}`);
  };

  render() {
    return (
      <Grid>
        <Grid.Row>
          <Image
            className="pokemon-logo"
            src={require("../../assets/pokemon_logo.png")}
          />
        </Grid.Row>
        <Grid.Row>
          {this.state.pokemons.map((pokemon, index) => {
            return (
              <Grid.Column width={5} key={index}>
                <Card onClick={() => this.handlePokemonSelect(pokemon)}>
                  <Image src={pokemon.image_url} />
                  <Card.Content>
                    <Card.Header>{pokemon.name}</Card.Header>
                    <Card.Description>
                      <List>
                        {pokemon.types.map(type => {
                          return <List.Item>{type.description}</List.Item>;
                        })}
                      </List>
                    </Card.Description>
                  </Card.Content>
                </Card>
              </Grid.Column>
            );
          })}
        </Grid.Row>
      </Grid>
    );
  }
}

export default PokeDex;
