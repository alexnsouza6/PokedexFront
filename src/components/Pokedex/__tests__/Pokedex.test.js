import React from "react";
import renderer from "react-test-renderer";
import { PokeDex } from "../PokeDex";
import { MemoryRouter as Router, withRouter } from "react-router-dom";

it("renders correctly", () => {
  const props = {
    pokemons: {
      byID: []
    },
    fetchPokemons: () => {}
  };
  const tree = renderer
    .create(
      <Router>
        <PokeDex {...props} />
      </Router>
    )
    .toJSON();
  expect(tree).toMatchSnapshot();
});
