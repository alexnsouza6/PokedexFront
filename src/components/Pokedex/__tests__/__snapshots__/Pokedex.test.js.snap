// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  className="ui grid"
>
  <div
    className="row"
  >
    <img
      className="ui image pokemon-logo"
      src="pokemon_logo.png"
    />
  </div>
  <a
    href="/create-pokemon"
    onClick={[Function]}
  >
    <button
      className="ui button"
      onClick={[Function]}
    >
      Create a pokemon
    </button>
  </a>
  <div
    className="ui icon input"
  >
    <input
      onChange={[Function]}
      placeholder="Search..."
      type="text"
      value=""
    />
    <i
      aria-hidden="true"
      className="search circular inverted link icon"
      onClick={[Function]}
    />
  </div>
  <div
    className="row"
  />
</div>
`;
